### PR TITLE
DowloadStreaming API for export instead of OpenReadAsync that buffers

### PR DESCRIPTION
**Pull request by smithago**
## Description
Testing perf improvements

## Related issues
Addresses [issue #].

## Testing
Describe how this change was tested.

### CODE REVIEW

File: src/Microsoft.Health.Dicom.Blob/Features/Storage/BlobFileStore.cs [hunk 1]
```diff
@@ -142,6 +142,26 @@ public class BlobFileStore : IFileStore
...
+        EnsureArg.IsNotNull(versionedInstanceIdentifier, nameof(versionedInstanceIdentifier));
+
+        BlockBlobClient blobClient = GetInstanceBlockBlobClient(versionedInstanceIdentifier, _blobMigrationFormatType);
+
+        Stream stream = null;
+        var blobOpenReadOptions = new BlobOpenReadOptions(allowModifications: false);
+
+        await ExecuteAsync(async () =>
+        {
+            Response<BlobDownloadStreamingResult> result = await blobClient.DownloadStreamingAsync(range: default, conditions: null, rangeGetContentHash: false, cancellationToken);
```
Comment (wsugarman) [inline src/Microsoft.Health.Dicom.Blob/Features/Storage/BlobFileStore.cs:158]: I know we had a number of conversations during the hackathon surrounding the various blob APIs we could use during export.  Has something changed, or could we always simply stream the blob to the destination without buffering?

File: src/Microsoft.Health.Dicom.Core/Features/Common/IFileStore.cs [hunk 2]
```diff
@@ -80,4 +80,14 @@ public interface IFileStore
...
         VersionedInstanceIdentifier versionedInstanceIdentifier,
         CancellationToken cancellationToken = default);
+
+    /// <summary>
+    /// Asynchronously gets a streaming file from the file store.
+    /// </summary>
+    /// <param name="versionedInstanceIdentifier">The DICOM identifier.</param>
+    /// <param name="cancellationToken">The cancellation token.</param>
+    /// <returns>A task that represents the asynchronous get operation.</returns>
+    Task<Stream> GetStreamingFileAsync(
```
Comment (wsugarman) [inline src/Microsoft.Health.Dicom.Core/Features/Common/IFileStore.cs:90]: Should we double-down on the "file" aspect and make this a method like `OpenReadAsync`?

**Review by wsugarman — APPROVED** [review]

File: src/Microsoft.Health.Dicom.Core/Features/Common/IFileStore.cs (hunk 2 above)
Comment (smithago) [reply src/Microsoft.Health.Dicom.Core/Features/Common/IFileStore.cs:90]: Probably. Right now kept it in the same pattern as the rest of the method names

File: src/Microsoft.Health.Dicom.Blob/Features/Storage/BlobFileStore.cs (hunk 1 above)
Comment (smithago) [reply src/Microsoft.Health.Dicom.Blob/Features/Storage/BlobFileStore.cs:158]: Not yet, we need to follow up with the Blob team to see, if the new SDK is available

**Pull request merged**